### ISSUE TITLE
Empty module exports in built output

### ISSUE THREAD

**Issue by m-mujica** [stealjs/steal#1305]
Sample code to replicate issue: https://github.com/asmarques1990/stealjs-antlr
Reported on https://gitter.im/stealjs/steal?at=5a0c31c671ad3f8736e8d293

while debugging this with @chasenlehara we noticed the problem is in this module:

![screen shot 2017-11-15 at 10 54 06](https://user-images.githubusercontent.com/724877/32852294-ae60c34e-c9f4-11e7-9ec5-7c95a19d2a96.png)

`require('./RuleContext')` returns an empty object when it should not.

The weird thing is that `define` is able to load `'./RuleContext'` correctly (see line 3345) [`arguments[4]` has the things exported by the module]

If I step into that `require` it goes through this code

![screen shot 2017-11-15 at 10 45 01](https://user-images.githubusercontent.com/724877/32852316-c2aedeb2-c9f4-11e7-8020-c4f5a8ff62e2.png)

But `entry.module.exports` is still empty after the link happens.

**Comment by matthewp** [comment]
sure this isn't circular deps?

**Comment by m-mujica** [comment]
I did not check for that. Let me see.

**Comment by m-mujica** [comment]
Yup, so `ParserRuleContext` -> `RuleContext` -> `antlr/tree/Trees` -> `ParserRuleContext`

**Comment by m-mujica** [comment]
Also, `RuleContext` -> `tree/Trees` -> `RuleContext`

![screen shot 2017-11-15 at 11 08 02](https://user-images.githubusercontent.com/724877/32852786-46d0a9cc-c9f6-11e7-8e31-07df109784a1.png)

![screen shot 2017-11-15 at 11 14 57](https://user-images.githubusercontent.com/724877/32852801-52ce03fa-c9f6-11e7-97bf-84f40dff400c.png)

**Comment by asmarques1990** [comment]
Thanks Guys! I'll open a PR to ANTLR4 with the fix. Meanwhile there is some workaround in StealJS?

**Comment by m-mujica** [comment]
@asmarques1990  We don't know if this is a bug in Steal.js yet. If you can reduce it down for us that would be awesome. You could either:

a) try to load antlr in Node.js and see if it works there. If the circular dependency is not a problem in Node.js, we have a bug.

b) Instead of testing in node you might also use something like webpack if it works there, then we need to figure out why it does not work in Steal.
